rust/Dockerfile-gateway

```diff
@@ -0,0 +1,11 @@
+FROM rust:1.70-slim as BUILDER
+WORKDIR /build/
+COPY . ./
+RUN cargo build -p gateway --release
```

Comment:
I'd strongly recommend to use a run cache here otherwise building these locally is an absolute pain. See https://docs.docker.com/build/cache/#use-the-dedicated-run-cache.